compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3466,6 +3467,23 @@ boolean checkLiteralAssignabilityBasedOnType(BLangLiteral baseLiteral, BLangLite
         return false;
     }
 
+    boolean validateFloatLiteral(Location pos, String numericLiteral) {
+        double value = Double.parseDouble(numericLiteral);
+        if (Double.isInfinite(value)) {
+            dlog.error(pos, DiagnosticErrorCode.FLOAT_TOO_LARGE, numericLiteral);
+            return false;
+        }
+
+        Pattern floatLiteral = Pattern.compile("[1-9]");
+        Pattern floatLiteralWithExpIndicator = Pattern.compile("([1-9])(.*)([eEpP])");
```

Comment:
Why we had to check two patterns?

Reply:
Removed the regex and refactor the previous logic